# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/CommandEncoder.cpp
# Commit: 430cadb32224
# Full Hash: 430cadb32224f974a8488db48bae07903ea63817
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-08-23 09:17:02
# Description:
#   Bug 1983820 - fix(webgpu): coerce `GPUTexture`s in `GPURenderPassDescriptor.depthStencilAttachment` to views properly r=webgpu-reviewers,teoxoy
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D261679
# ==============================================================================

diff -r 488f7a910fe4 -r 430cadb32224 dom/webgpu/CommandEncoder.cpp
--- a/dom/webgpu/CommandEncoder.cpp	Sat Aug 23 01:42:10 2025 +0000
+++ b/dom/webgpu/CommandEncoder.cpp	Sat Aug 23 03:24:26 2025 +0000
@@ -221,32 +221,36 @@
     const dom::GPURenderPassDescriptor& aDesc) {
   dom::GPURenderPassDescriptor desc{aDesc};
 
+  auto coerceToViewInPlace =
+      [](dom::OwningGPUTextureOrGPUTextureView& texOrView)
+      -> RefPtr<TextureView> {
+    RefPtr<TextureView> view;
+    switch (texOrView.GetType()) {
+      case dom::OwningGPUTextureOrGPUTextureView::Type::eGPUTexture: {
+        dom::GPUTextureViewDescriptor defaultDesc{};
+        RefPtr<Texture> tex = texOrView.GetAsGPUTexture();
+        texOrView.SetAsGPUTextureView() = tex->CreateView(defaultDesc);
+        break;
+      }
+
+      case dom::OwningGPUTextureOrGPUTextureView::Type::eGPUTextureView:
+        // Nothing to do, great!
+        break;
+    }
+    view = texOrView.GetAsGPUTextureView();
+    return view;
+  };
+
   for (auto& at : desc.mColorAttachments) {
-    auto coerceToViewInPlace =
-        [](dom::OwningGPUTextureOrGPUTextureView& texOrView)
-        -> RefPtr<TextureView> {
-      RefPtr<TextureView> view;
-      switch (texOrView.GetType()) {
-        case dom::OwningGPUTextureOrGPUTextureView::Type::eGPUTexture: {
-          dom::GPUTextureViewDescriptor defaultDesc{};
-          RefPtr<Texture> tex = texOrView.GetAsGPUTexture();
-          texOrView.SetAsGPUTextureView() = tex->CreateView(defaultDesc);
-          break;
-        }
-
-        case dom::OwningGPUTextureOrGPUTextureView::Type::eGPUTextureView:
-          // Nothing to do, great!
-          break;
-      }
-      view = texOrView.GetAsGPUTextureView();
-      return view;
-    };
     TrackPresentationContext(coerceToViewInPlace(at.mView)->GetTargetContext());
     if (at.mResolveTarget.WasPassed()) {
       TrackPresentationContext(
           coerceToViewInPlace(at.mResolveTarget.Value())->GetTargetContext());
     }
   }
+  if (desc.mDepthStencilAttachment.WasPassed()) {
+    coerceToViewInPlace(desc.mDepthStencilAttachment.Value().mView);
+  }
 
   RefPtr<RenderPassEncoder> pass = new RenderPassEncoder(this, desc);
   pass->SetLabel(desc.mLabel);
